tutorials/k8s-kapsule-multi-az/index.mdx

```diff
@@ -0,0 +1,625 @@
+
+1. For this new section, create a new Terraform file `elasticsearch.tf`. Deploying a clustered application commonly implies a more complex deployment logic. For Elasticsearch and Kibana, this logic is managed by ECK, the Elasticsearch operator for Kubernetes. It will be in charge of creating Elasticsearch nodes and configuring them to make a cluster. It will also automatically configure Kibana to connect to the Elasticsearch cluster. So the first thing to do is to deploy the ECK operator using its chart with the Helm provider.
+    ```
+    resource "helm_release" "eck" {
+    name      = "elastic-operator"
+    namespace = "elastic-system"
+
+    create_namespace = true
+
+    repository = "https://helm.elastic.co"
+    chart      = "eck-operator"
+
+    depends_on = [helm_release.nginx_ingress]
+    }
+    ```
+
+2. Now we create a namespace to deploy your Elasticsearch cluster and Kibana instance. We use the `depends_on` parameter to make sure that the namespace is created after the ECK operator, so the Elasticsearch and Kibana CRDs are available before we try to create them.
+    ```
+    resource "kubernetes_namespace" "elasticsearch" {
+    metadata {
+        name = "elasticsearch"
+    }
+
+    depends_on = [helm_release.eck]
+    }
+    ```
+
+3. To create your multi-AZ Elasticsearch cluster, we will use the "Elasticsearch" CRD made available by the ECK operator. With it, we can now deploy an Elasticsearch cluster via a `nodeSets` with three Elasticsearch nodes, one in each AZ, with a persistent volume bound to each Elasticsearch nodes to persists data. We have to use different parameters to spread the pods across the different AZs. The first one is to set the `storageClassName` to `scw-bssd`, allowing the creation to the correct AZ of the Elasticsearch volumes as explained before. Adding the annotation `eck.k8s.elastic.co/downward-node-labels` we ask the operator to set the label `topology.kubernetes.io/zone` on each created pod, which will be used as an environment variable to set the `node.attr.zone` parameter on each Elasticsearch node. This allows Elasticsearch to allocate primary and replica shards to pods running on different AZs and nodes. Finally, we set the `topologySpreadConstraints` to spread the pods across the different AZs based on the `topology.kubernetes.io/zone` label defined on the nodes by Kapsule. If we have three node pools, one in each AZ, and nodeset `count` of 3, we will have one pod per AZ. So in case of loss of one AZ, we will still have two pods running in the other AZs and the Elasticsearch cluster will still be available, maintaining the quorum of 2 nodes needed to elect a master node.
```

Comment:
Same as comment above: I suggest to break this huge chunk into sub pieces, for example:
- The first one is to...
- Adding the annotation ...
- Finally, we set...

Reply:
```suggestion
3. To create your multi-AZ Elasticsearch cluster, we will use the "Elasticsearch" CRD provided by the ECK operator.

Then deploy an Elasticsearch cluster using `nodeSets`, consisting of three Elasticsearch nodes, one in each AZ. These nodes will have persistent volumes to store data.

Next, configure various parameters to ensure pod distribution across AZs. 

First, start the `storageClassName` to `scw-bssd`. This choice ensures that Elasticsearch volumes are correctly created in the respective AZs.

Then add the annotation `eck.k8s.elastic.co/downward-node-labels`. This annotation instructs the operator to assign the label `topology.kubernetes.io/zone` to each pod it creates. This label serves as an environment variable to set the `node.attr.zone` parameter on each Elasticsearch node. This step enables Elasticsearch to allocate primary and replica shards to pods in different AZs and nodes.

Finally, configure `topologySpreadConstraints` to distribute pods across AZs based on the `topology.kubernetes.io/zone` label defined on the nodes by Kapsule. For instance, if there are three node pools, each in a different AZ, and a nodeset `count` of 3, this configuration will ensure one pod per AZ. This redundancy ensures that even if one AZ experiences a failure, two pods will continue to run in the other AZs, maintaining the necessary quorum of 2 nodes needed to elect a master node and keeping the Elasticsearch cluster available.
```

@Nox-404 wdyt?

Reply:
I'm not sure that the "Next, configure various parameters to ensure pod distribution across AZs" is useful, it doesn't seem to be a step